### PR TITLE
fix:proto nesting causes template generation problems 

### LINKED ISSUE #2717: proto nesting causes template generation problems
#### What happened:
```pb

message CreateTemplate {
    message Request {
        string template_name = 1;
        string bind_interface_key = 2;
        map<string, string> meta = 3;
    }
    message Response {
        message Data {
            int64 template_id = 1;
        }
        int64 code = 1;
        string message = 2;
        Data data = 3;
    }
}

message GetTemplates {
    message Request {
        int64 page = 1;
        int64 page_size = 2;
    }

    message Response {
        message Data {
            int64 template_id = 1;
            string template_name = 2;
        }
        int64 code = 1;
        string message = 2;
        Data data = 3;
    }
}
```

generate server
```
func (s *TemplateService) CreateTemplate(ctx context.Context, req *pb.CreateTemplate.Request) (*pb.CreateTemplate.Response, error) {
	return &pb.CreateTemplate.Response{}, nil
}
func (s *TemplateService) GetTemplates(ctx context.Context, req *pb.GetTemplates.Request) (*pb.GetTemplates.Response, error) {
	return &pb.GetTemplates.Response{}, nil
}
func (s *TemplateService) GetTemplateById(ctx context.Context, req *pb.GetTemplateById.Request) (*pb.GetTemplateById.Response, error) {
	return &pb.GetTemplateById.Response{}, nil
}
func (s *TemplateService) SaveTemplateById(ctx context.Context, req *pb.SaveTemplateById.Request) (*pb.SaveTemplateById.Response, error) {
	return &pb.SaveTemplateById.Response{}, nil
}
func (s *TemplateService) DeleteTemplateById(ctx context.Context, req *pb.DeleteTemplateById.Request) (*pb.DeleteTemplateById.Response, error) {
	return &pb.DeleteTemplateById.Response{}, nil
}

...


```

#### Environment:
- Kratos version (use `kratos -v`):
kratos version v2.6.1
- Go version (use `go version`):
1.19
- OS (e.g: `cat /etc/os-release`):
mac
- Others:

### PR DESCRIPTION
**Pull request by songzhibin97**
fix #2717

### CODE REVIEW

**Comment by songzhibin97** [comment]
```

func (s *TemplateService) CreateTemplate(ctx context.Context, req *pb.CreateTemplate_Request) (*pb.CreateTemplate_Response, error) {
	return &pb.CreateTemplate_Response{}, nil
}
func (s *TemplateService) GetTemplates(ctx context.Context, req *pb.GetTemplates_Request) (*pb.GetTemplates_Response, error) {
	return &pb.GetTemplates_Response{}, nil
}
func (s *TemplateService) GetTemplateById(ctx context.Context, req *pb.GetTemplateById_Request) (*pb.GetTemplateById_Response, error) {
	return &pb.GetTemplateById_Response{}, nil
}
func (s *TemplateService) SaveTemplateById(ctx context.Context, req *pb.SaveTemplateById_Request) (*pb.SaveTemplateById_Response, error) {
	return &pb.SaveTemplateById_Response{}, nil
}
func (s *TemplateService) DeleteTemplateById(ctx context.Context, req *pb.DeleteTemplateById_Request) (*pb.DeleteTemplateById_Response, error) {
	return &pb.DeleteTemplateById_Response{}, nil
}
```

**Comment by shenqidebaozi** [comment]
help supplement the unit test

File: cmd/kratos/internal/proto/client/client.go [hunk 1]
```diff
@@ -16,8 +16,8 @@ import (
 // CmdClient represents the source command.
 var CmdClient = &cobra.Command{
 	Use:   "client",
-	Short: "Generate the proto client code",
-	Long:  "Generate the proto client code. Example: kratos proto client helloworld.proto",
+	Short: "Generate the proto Client code",
+	Long:  "Generate the proto Client code. Example: kratos proto client helloworld.proto",
```
Comment (haiyux) [inline cmd/kratos/internal/proto/client/client.go:20]: Why change to uppercase?

Comment (songzhibin97) [reply cmd/kratos/internal/proto/client/client.go:20]: Smooth and consistent with Server

Comment (haiyux) [reply cmd/kratos/internal/proto/client/client.go:20]: This is just a hint. `use` is  `client`?

**Review by haiyux — APPROVED** [review]

**Review by shenqidebaozi — APPROVED** [review]

**Comment by fttx2020** [comment]
this fix will cause google.protobuf.Empty parse failed
please fix it!
source file:kratos/cmd/kratos/internal/proto/server/template.go
lines:120
code: const empty = "google.protobuf.Empty"
expected to: const empty = "google_protobuf_Empty"

**Comment by songzhibin97** [comment]
> this fix will cause google.protobuf.Empty parse failed please fix it! source file:kratos/cmd/kratos/internal/proto/server/template.go lines:120 code: const empty = "google.protobuf.Empty" expected to: const empty = "google_protobuf_Empty"

后面好像有pr已经fix了

**Comment by henryjhenry** [comment]
> > this fix will cause google.protobuf.Empty parse failed please fix it! source file:kratos/cmd/kratos/internal/proto/server/template.go lines:120 code: const empty = "google.protobuf.Empty" expected to: const empty = "google_protobuf_Empty"
> 
> 后面好像有pr已经fix了

目前还存在这个问题
kratos: 2.7.3

**Pull request merged**